### ISSUE TITLE
mame 0.78 drivers with mame2003-plus core

### ISSUE THREAD

**Issue by markwkidd** [markwkidd/mame2003-legacy-libretro#3]
I have created this fork of the mame2003 codebase to bring the mame2003-plus "core" code  -- generally its mame functions and the libretro port -- together with the MAME 0.78 drivers.

During the heavy development of mame2003-plus it wasn't practical to keep these the two forks in sync with git but I believe it would be now, and would be a useful ting.

The purposes of this approach would be to bring additional features from mame2003-plus to a version of the core that only has the MAME 0.78 drivers. With the two codebases synchronized this way, it will be possible to cherry-pick future "core" commits from mame2003-plus and bring them to the MAME 0.78 drivers in a mostly-automated way.

Ultimately I would like to only have one mame2003. This has nothing to do with that ultimate goal and doesn't really advance it.

**Comment by markwkidd** [comment]
@grant2258 this is what I have been cooking up. It currently compiles in Windows 10 x64 and runs Defender.

**Comment by grant2258** [comment]
makes sense now I thought you were just giving yourself hard work, Me well I would be lazy id copy all the core files hehe

**Comment by markwkidd** [comment]
The first time I did this project I started with the mame2003 repository and I copied the Plus core files over, then fixed everything.

I made a couple of messy mistakes that I was contemplating going back to clean up when I realized that by starting with mame2003's repo I lost the commit history of mame2003-plus repository, which is the opposite of what I want for using git to keep things in sync. Oops.

So I started again with the Plus repository which has all of the mame2003 commits as well as our new ones. It was faster to do the porting work this time. Only five or six hours compared to much longer the first time that ended in failure. And the result is much cleaner. That's how it goes. :)

In terms of manual porting work, I had to work through some of the vector code, hiscore support, things like that. A few of the more complex projects in the mame2003 coding hall of fame but nothing too bad. Minor fixes that were mostly just adapting function calls or variable names.
